### DISCUSSION TITLE
Inyect services

### DISCUSSION THREAD

**Discussion by psjavi** [rappasoft/laravel-livewire-tables#1740] [Q&A]
Hi, I usually use Services and I inyected them in Livewire as following:

`    private CompanyService $companyService;
    private DepartmentService $departmentService;
    private RoleService $roleService;

    public function boot(
        CompanyService    $companyService,
        DepartmentService $departmentService,
        RoleService       $roleService
    ): void
    {
        $this->companyService = $companyService;
        $this->departmentService = $departmentService;
        $this->roleService = $roleService;
    }
`

How can I do it in the TableComponent?

Thanks.

**Comment by lrljoe** [comment]
If you're using Services in that way, then it'd be smoother to singleton it!

**Reply by psjavi** [reply]
How should I go about it?
